src/core/server/plugins/plugins_system.test.ts

```diff
@@ -435,6 +436,21 @@ describe('setup', () => {
       `[Error: Setup lifecycle of "timeout-setup" plugin wasn't completed in 30sec. Consider disabling the plugin and re-start.]`
     );
   });
+
+  it('logs only server-side plugins', async () => {
+    [
+      createPlugin('order-0'),
+      createPlugin('order-not-run', { server: false }),
+      createPlugin('order-1'),
+    ].forEach((plugin, index) => {
+      jest.spyOn(plugin, 'setup').mockResolvedValue(`setup-as-${index}`);
+      jest.spyOn(plugin, 'start').mockResolvedValue(`started-as-${index}`);
```

Comment:
Are these `jest.spyOn` calls necessary?

Reply:
Yes unforuntately, the PluginWrapper methods end up trying to require code that doesn't exist, so they must be stubbed out. I initially went down the path of creating a full mock for this class but it ended up being quite cumbersome due to the number of public properties exposed and generics involved. Didn't seem worth the effort right now.